docs/rfc/active/graph_ql_frontend/README.md

```diff
@@ -0,0 +1,607 @@
+
+```java
+/*
+ * Build DataLoader Registry
+ */
+DataLoaderRegistry registry = new DataLoaderRegistry();
+registry.register("datasetLoader", datasetLoader);
+registry.register("corpUserLoader", corpUserLoader);
+
+/*
+ * Construct execution input
+ */
+ExecutionInput executionInput = ExecutionInput.newExecutionInput()
+.query(queryJson.asText())
+.variables(variables)
+.dataLoaderRegistry(register)
+.build();
+
+/*
+ * Execute GraphQL Query
+ */
+ExecutionResult executionResult = _engine.execute(executionInput);
+```
+
+For more information about `DataLoaders` see the [Using DataLoader](https://www.graphql-java.com/documentation/v15/batching/) doc. 
+
+For a full reference implementation of the query execution process, see the  ``GraphController.java`` class associated with this PR. 
+
+
+### Onboarding GMA Models
+
+The first phase of the GQL rollout will support primary-key lookups of GMA entities and projection of their associated aspects. In order to achieve this, we will
+- model entities, aspects, and the relationships among them
+- expose queries against top-level entities
+using GQL. 
+
+The proposed steps for onboarding a GMA entity, it's related aspects, and the relationships among them will be outlined next. 
+
+####1. **Model an entity in GQL**
+
+The entity model should include its GMA aspects, as shown below, as simple object fields. The client will need not be intimately aware of the concept of "aspects". Instead, it should be concerned with entities and their relationships.
+
+*Modeling Dataset*
+```javascript
+"""
+Represents the GMA Dataset Entity
+"""
+type Dataset {
+
+    urn: String!
+
+    platform: String!
+
+    name: String!
+
+    origin: FabricType
+
+    description: String
+
+    uri: String
+
+    platformNativeType: PlatformNativeType
+
+    tags: [String]!
+
+    properties: [PropertyTuple]
+
+    createdTime: Long!
+
+    modifiedTime: Long!
+
+    ownership: Ownership
+}
+
+"""
+Represents Ownership
+"""
+type Ownership {
+
+    owners: [Owner]
+
+    lastModified: Long!
+}
+
+"""
+Represents an Owner
+"""
+type Owner {
+    """
+     The fully-resolved owner
+    """
+    owner: CorpUser!
+
+    """
+     The type of the ownership
+    """
+    type: OwnershipType
+
+    """
+     Source information for the ownership
+    """
+    source: OwnershipSource
+}
+```
+
+Notice that the Dataset's ``Ownership`` aspect includes a nested ``Owner`` field that references a ``CorpUser`` type. In the GMS model, this is represented as a foreign-key relationship (urn). In GraphQL, we include the *resolved* relationship, allowing the client to easily retrieve information about a Dataset's owners.
+
+To support traversal of this relationship, we additionally include a ``CorpUser`` type:
+
+```javascript
+"""
+Represents the CorpUser GMA Entity
+"""
+type CorpUser {
+
+    username: String!
+
+    info: CorpUserInfo
+
+    editableInfo: CorpUserEditableInfo
+}
+```
+
+####2. **Extend the 'Query' type**: 
```

Comment:
This doesn't render quite right, I think you need a space after the `#`

Reply:
Thanks :)